src/components/Common/SearchBarDropdown.tsx

```diff
@@ -0,0 +1,207 @@
+import { useNavigate } from "react-router-dom";
+
+type SearchBarDropDownProps = {
+  placeholder: string;
+};
+
+const styles = {
+  autocomplete: {
+    "&.MuiAutocomplete-hasClearIcon .MuiOutlinedInput-root": { paddingX: 1 },
+    position: "relative",
+    padding: 0,
+  },
+  box: {
+    paddingBottom: 1,
+    paddingTop: 0,
+    paddingX: 2,
+  },
+  listBox: {
+    maxHeight: "60vh",
+  },
+  searchIcon: {
+    paddingRight: 0,
+    "&.Mui-disabled": {
+      color: "gray.main",
+    },
+  },
+  searchIconFocused: {
+    paddingRight: 0,
+    "&.Mui-disabled": {
+      color: "primary.main",
+    },
+  },
+};
+
+const SearchBarDropDown = ({ placeholder }: SearchBarDropDownProps) => {
+  const [focusSearchBar, setFocusSearchBar] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [keywords, setKeywords] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [searchOptions, setSearchOptions] = useState<SelectedUser[]>([]);
+  const { getAccessTokenSilently } = useAuth0();
+  const navigate = useNavigate();
+  const debouncedFetch = useMemo(
+    () => debounce((keywords: string) => fetchUsers(keywords), 100),
+    []
+  );
+
+  const fetchUsers = async (keywords: string) => {
+    setLoading(true);
+    if (keywords.length === 0) {
+      return;
+    }
+
+    try {
+      const token = await getAccessTokenSilently();
+      const result = await axios.get(
+        `http://localhost:3001/api/users/searchUsers`,
+        {
+          params: {
+            keyword: keywords,
+          },
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      );
+
+      setSearchOptions(result.data as SelectedUser[]);
+
+      if (keywords.trim() === "") {
+        setSearchOptions([]);
+      }
+    } catch (error) {
+      console.error("Failed to fetch users:", error);
+      setSearchOptions([]);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    debouncedFetch(keywords);
+  }, [keywords]);
+
+  const handleInputChange = (newInputValue: string) => {
+    setKeywords(newInputValue);
+
+    if (newInputValue.trim() === "") {
+      setSearchOptions([]);
+    }
+  };
+
+  const handleSelect = (selectedUsername: string) => {
+    setKeywords("");
+    setSearchOptions([]);
+    inputRef.current?.blur();
+    navigate(`/${selectedUsername}`);
+  };
+
+  const handleClear = () => {
+    setKeywords("");
+    setSearchOptions([]);
+    setFocusSearchBar(false);
+  };
+
+  return (
+    <Box sx={styles.box}>
+      <Autocomplete
+        disablePortal
+        fullWidth
+        filterOptions={(x) => x}
+        getOptionLabel={() => ""}
+        forcePopupIcon={false}
+        id="search"
+        value={null}
+        inputValue={keywords}
+        ListboxProps={{ style: styles.listBox }}
+        loading={!keywords && loading}
+        loadingText="Start typing to search..."
```

Comment:
This is a minor thing, but it looks like we're mixing a few things up here. I think the `loading` prop shouldn't depend on `!keywords`. Also, the loading text should be something like "Loading..." (which is the default), since it should show when the fetch is occurring (ie. the user has already typed something in)

The other side of this is when the user hasn't typed anything in, in which case we should be showing "Start typing to search...", or when there are no results found, in which case we should show "No user found".

So, I feel like we should change these lines to handle only loading stuff and move the no options logic into `noOptionsText` like so:

```
noOptionsText={!keywords ? "Start typing to search...": "No user found"}
```

I tried it out an ran into another complication where the keywords gets set before `fetchUsers` runs, so this makes our `noOptionsText` flash for a split second before the loading state is set. So, we may have to set the loading state true in `handleInputChange` or something like that

Reply:
Cool so if I follow this method, there is a case of the loading state showing up for half a second or so due to loading with every keyword. I could void the loading state from the autocomplete in general because we do take care of it within fetchUsers as well as debouncing right?